src/pytypes/onefuzztypes/requests.py

```diff
@@ -91,6 +91,11 @@ class PoolCreate(BaseRequest):
     arch: Architecture
     managed: bool
     client_id: Optional[UUID]
+    max_size: int
```

Comment:
These need to be in an optional sub-struct for a few reasons.

1. We want to use pools for compute resources that are not Azure VM Scalesets
2. This forces auto-resizing where there are multiple cases where manually managed scalesets are preferred
3. Backwards compatibility